types.go

```diff
@@ -1770,7 +1773,16 @@ type Message struct {
 // more complex usages.
 func NewMessage(data []byte) *Message {
 	return &Message{
-		Data: [][]byte{data},
+		Data:       [][]byte{data},
+		doneSignal: make(chan struct{}),
+	}
+}
+
+// done closes the internal doneSignal channel to let the receiver know that this message has been acted upon
+func (m *Message) done() {
+	// TODO: move initialization in ctor and use ctor everywhere?
```

Comment:
I decided to always initialize it before passing the message down to the handlers instead. thoughts?